src/uu/sort/src/sort.rs

```diff
@@ -1085,6 +1086,160 @@ fn get_rlimit() -> UResult<usize> {
 }
 
 const STDIN_FILE: &str = "-";
+
+/// Legacy `+POS1 [-POS2]` syntax is permitted unless `_POSIX2_VERSION` is in
+/// the [TRADITIONAL, MODERN) range (matches GNU behaviour).
+fn allows_traditional_usage() -> bool {
+    !matches!(uucore::posix::posix_version(), Some(ver) if (TRADITIONAL..MODERN).contains(&ver))
+}
+
+#[derive(Debug, Clone)]
+struct LegacyKeyPart {
+    field: usize,
+    char: usize,
+    opts: String,
+}
+
+fn parse_usize_or_max(num: &str) -> Option<usize> {
+    match num.parse::<usize>() {
+        Ok(v) => Some(v),
+        Err(e) if *e.kind() == IntErrorKind::PosOverflow => Some(usize::MAX),
+        Err(_) => None,
+    }
+}
+
+fn parse_legacy_part(spec: &str, default_char: usize) -> Option<LegacyKeyPart> {
+    let mut idx = 0;
+    let bytes = spec.as_bytes();
+    while idx < bytes.len() && bytes[idx].is_ascii_digit() {
+        idx += 1;
+    }
```

Comment:
I think a functional approach is cleaner. Something like:
```suggestion
    let idx = spec.chars().take_while(|c| c.is_ascii_digit()).count();
```

Reply:
This was fixed in this commit. 3b52b5d8ad147bb0cc17f08da4d1de9519512a25